pyscriptjs/src/python/pyscript.py

```diff
@@ -166,7 +180,7 @@ def write(self, value, append=False):
         if mime_type in ("application/javascript", "text/html"):
             script_element = document.createRange().createContextualFragment(html)
             out_element.appendChild(script_element)
-        else:
+        elif hasattr(out_element, "innerHTML"):
             out_element.innerHTML = html
```

Comment:
this look like a code smell. Why do we need to check whether `out_element` has `"innerHTML"`?

Reply:
(I think that the code is outdate, but answering just for the sake of conversation)

> I think it's the result of some previous behavior that I can't reproduce anymore.

if you can kill a line of code without causing any fail to test, it either means that the line of code is useless (and so it should be removed) or tests are missing (and so they should be added).

> But, why is it a code smell?

`out_element` is supposed to be an  instance of `Element` and as such it always have the `innerHTML` property. If it doesn't it means that we are passing something of the wrong type, but then this is the wrong place where to fix the problem: it should be fixed earlier and avoiding passing an `out_element` which is not actually an element.